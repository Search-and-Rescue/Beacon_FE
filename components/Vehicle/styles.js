const styles = {
  vehicleContainer: {
    flex: 1,
    marginVertical: 40,
    alignItems: "center"
  },
  label: {
    fontSize: 12,
    textAlign: "left",
    width: "80%"
  },
  input: {
    backgroundColor: "#fff",
    borderColor: "black",
    borderWidth: 1,
    marginBottom: 10,
    padding: 10,
    width: "80%"
  },
  updateBtn: {
    borderColor: "black",
    borderWidth: 1,
    padding: 10,
    width: "80%"
  },
  updateBtnText: {
    textAlign: "center"
  }
};

export default styles;