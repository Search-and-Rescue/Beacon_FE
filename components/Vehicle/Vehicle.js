import React, { Component } from "react";
import { View, Text, TextInput, TouchableOpacity } from "react-native";
import { addVehicle, getVehicles } from '../../util/apiCalls';
import { connect } from "react-redux";
import styles from "./styles";
import { setVehicles } from '../../actions';
import { bindActionCreators } from 'redux';

class Vehicle extends Component {
  constructor(props) {
    super(props);
    if (this.props.navigation.state.params === undefined) {
      this.props.navigation.state.params = {vehicle: {}}
    }
    const { vehicle } = this.props.navigation.state.params;
    this.state = {
      make: vehicle.make || "",
      model: vehicle.model || "",
      year: vehicle.year || "",
      color: vehicle.color || "",
      licensePlate: vehicle.licensePlate || "",
      state: vehicle.state || ""
    };
  }

  handleSubmit = async () => {
    const { navigation, user } = this.props;
    const newVehicle = {
      ...this.state,
      userId: parseInt(user.id)
    }
    await addVehicle(newVehicle);
    const userInfoVehicles = await getVehicles();
    await this.props.setVehicles(userInfoVehicles.user.vehicles);
    navigation.navigate("VehicleList");
    this.setState({
      make: "",
      model: "",
      year: "",
      color: "",
      licensePlate: "",
      state: ""
    });
  }

  render() {
    return (
      <View style={styles.vehicleContainer}>
        <Text>Make</Text>
        <TextInput
          placeholder="make"
          style={styles.input}
          onChangeText={text => this.setState({ make: text })}
          value={this.state.make}
          name="make"
        />
        <Text>Model</Text>
        <TextInput
          placeholder="model"
          style={styles.input}
          onChangeText={text => this.setState({ model: text })}
          value={this.state.model}
          name="model"
        />
        <Text>Year</Text>
        <TextInput
          placeholder="year"
          style={styles.input}
          onChangeText={text => this.setState({ year: text })}
          value={String(this.state.year)}
          name="year"
        />
        <Text>Color</Text>
        <TextInput
          placeholder="color"
          style={styles.input}
          onChangeText={text => this.setState({ color: text })}
          value={this.state.color}
          name="color"
        />
        <Text>License Plate</Text>
        <TextInput
          placeholder="license plate"
          style={styles.input}
          onChangeText={text => this.setState({ licensePlate: text })}
          value={this.state.licensePlate}
          name="license_plate"
        />
        <Text>License State</Text>
        <TextInput
          placeholder="license state"
          style={styles.input}
          onChangeText={text => this.setState({ state: text })}
          value={this.state.state}
          name="state"
        />
        <TouchableOpacity style={styles.updateBtn} onPress={this.handleSubmit}>
          <Text style={styles.updateBtnText}>Update</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

export const mapStateToProps = ({ user }) => ({
  user
});

export const mapDispatchToProps = dispatch => bindActionCreators({ setVehicles }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Vehicle);
