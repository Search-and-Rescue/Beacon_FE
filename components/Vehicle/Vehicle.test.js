import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import Vehicle from './Vehicle';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Vehicle', () => {

  beforeEach(() => {
    mockVehicle = {
      navigation: {
        state: {
          params: {
            vehicle: {
              id: 1,
              make: "Ford",
              model: "F150",
              year: 2010,
              color: "silver",
              licensePlate: "CYE 909"
            }
        }
      }
    }
  }
  });

  it('should match the snapshot with all of the data passed through', () => {
    expect(shallow(<Vehicle {...mockVehicle}/>)).toMatchSnapshot();
  });
});