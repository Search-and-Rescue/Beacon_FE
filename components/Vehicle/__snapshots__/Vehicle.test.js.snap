// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicle should match the snapshot with all of the data passed through 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <Text>
    Vehicle Page
  </Text>
  <TextInput
    allowFontScaling={true}
    name="make"
    onChangeText={[Function]}
    placeholder="make"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "black",
        "borderWidth": 1,
        "marginBottom": 10,
        "padding": 10,
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    name="model"
    onChangeText={[Function]}
    placeholder="model"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "black",
        "borderWidth": 1,
        "marginBottom": 10,
        "padding": 10,
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    name="year"
    onChangeText={[Function]}
    placeholder="year"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "black",
        "borderWidth": 1,
        "marginBottom": 10,
        "padding": 10,
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    name="color"
    onChangeText={[Function]}
    placeholder="color"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "black",
        "borderWidth": 1,
        "marginBottom": 10,
        "padding": 10,
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    name="license_plate"
    onChangeText={[Function]}
    placeholder="license plate"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "black",
        "borderWidth": 1,
        "marginBottom": 10,
        "padding": 10,
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 1,
        "padding": 10,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Update
    </Text>
  </TouchableOpacity>
</View>
`;
